CoreGraphNode/src/au/gov/asd/tac/constellation/graph/node/plugins/DefaultPluginInteraction.java

```diff
@@ -236,15 +236,19 @@ public boolean cancel() {
     }
 
     @Override
-    public boolean prompt(final String promptName, final PluginParameters parameters) {
+    public boolean prompt(final String promptName, final PluginParameters parameters, final boolean okButtonFocused) {
         if (SwingUtilities.isEventDispatchThread()) {
             throw new IllegalStateException("Plugins should not be run on the EDT!");
         }
 
         boolean result = false;
 
         final PluginParametersSwingDialog dialog = new PluginParametersSwingDialog(promptName, parameters);
-        dialog.showAndWait();
+        if (okButtonFocused) {
```

Comment:
I hate this so much but have no better solution to offer :\

My only comment is that i would do away with the `okButtonFocused` parameter - you're always using a mutant JavaFX/Swing dialog here so the problem will always be a problem so just hard-code the no focus behaviour in.

Reply:
So the only way I can think of to remove the parameters is to make `okButtonFocused` in `PluginExecution` static. While it would be good to call the variable statically, we don't really want it to behave statically (as changing the value for one plugin would affect all subsequent plugin calls). That will mean that we need to either change the value back to true at the end of `showAndWaitNoFocus()` or having to always specify the value of `okButtonFocused` anytime a plugin is run interactively via the PluginExecution.

Curious as to whether you think this is better than the parameters option @cygnus-x-1? Personally I this isn't going to look clean either way

Reply:
I'm not sure I follow... What I was trying to get at is that any time you run a plugin interactively, it's going to create a `PluginParametersSwingDialog` (which is JavaFX in Swing) and you'll have this problem, so why not just hard-code it's default behaviour to unfocus the button rather than ask the developer to specify?

Reply:
The reason for having the option is a UX thing. While there are cases for having the ok button unfocused, I think these cases are exceptions rather than the new standard. I think having the ok button focused is still preferred for most plugins that are run interactively and so I don't want to take that option away (I personally tried some plugins like sphere graph builder without ok enabled and found myself still wanting to hit enter to execute the plugin)

Reply:
Well in that case what you've done is fine. My issue with all of this is that it's yet another thing developers need to remember when developing plugins, and it only exists because we're not handling Swing/JavaFX integration properly rather than because of a design decision. This whole thing is a stop-gap measure and should be fixed properly.